steps/run/step.py

```diff
@@ -0,0 +1,43 @@
+import logging
+import json
+from relay_sdk import Interface, Dynamic as D
+
+relay = Interface()
+
+name = relay.get(D.name)
+
+params = None
+param_sets = None
```

Comment:
You could probably reduce the duplication of the logic below by just making the one-run case a special case of `param_sets`, i.e., instead of `params = relay.get(D.parameters)`, say `param_sets = [relay.get(D.parameters)]`. (Then change the log message based on `len(param_sets)` or whatever is convenient, as I think it's the only thing that's actually different between those code paths.